src/extensions/filter-control/bootstrap-table-filter-control.js

```diff
@@ -111,64 +121,77 @@ $.BootstrapTable = class extends $.BootstrapTable {
     // Make sure that the filterControl option is set
     if (this.options.filterControl) {
       // Make sure that the internal variables are set correctly
-      this.options.valuesFilterControl = []
+      this._valuesFilterControl = []
+      this._initialized = false
+      this._usingMultipleSelect = false
 
       this.$el
         .on('reset-view.bs.table', () => {
-          // Create controls on $tableHeader if the height is set
-          if (!this.options.height) {
-            return
-          }
-
-          // Avoid recreate the controls
-          const $controlContainer = UtilsFilterControl.getControlContainer(this)
-
-          if (
-            ($controlContainer.find('select').length > 0 || $controlContainer.find('input:not([type="checkbox"]):not([type="radio"])').length > 0) &&
-            !this.options.filterControlContainer
-          ) {
-            return
+          setTimeout(() => {
+            UtilsFilterControl.initFilterSelectControls(this)
+            UtilsFilterControl.setValues(this)
+          }, 2)
+        })
+        .on('toggle.bs.table', (_, cardView) => {
+          this._initialized = false
+          if (!cardView) {
+            UtilsFilterControl.initFilterSelectControls(this)
+            UtilsFilterControl.setValues(this)
+            this._initialized = true
           }
-
-          UtilsFilterControl.createControls(this, $controlContainer)
         })
         .on('post-header.bs.table', () => {
+          setTimeout(() => {
+            UtilsFilterControl.initFilterSelectControls(this)
+            UtilsFilterControl.setValues(this)
+            setTimeout(() => {
+              const container = UtilsFilterControl.getControlContainer(this)
+              const multipleSelects = container.find('.fc-multipleselect')
+
+              if (multipleSelects.length > 0) {
+                multipleSelects.multipleSelect('destroy').multipleSelect()
+              }
+            }, 2)
+          }, 2)
+        })
+        .on('column-switch.bs.table', () => {
           UtilsFilterControl.setValues(this)
+          if (this.options.height) {
+            this.fitHeader()
+          }
         })
         .on('post-body.bs.table', () => {
-          if (this.options.height && !this.options.filterControlContainer) {
+          if (this.options.height && !this.options.filterControlContainer && this.options.filterControlVisible) {
             UtilsFilterControl.fixHeaderCSS(this)
           }
           this.$tableLoading.css('top', this.$header.outerHeight() + 1)
         })
-        .on('column-switch.bs.table', () => {
-          UtilsFilterControl.setValues(this)
-        })
-        .on('load-success.bs.table', () => {
-          this.enableControls(true)
-        })
-        .on('load-error.bs.table', () => {
-          this.enableControls(true)
+        .on('all.bs.table', () => {
+          UtilsFilterControl.syncHeaders(this)
```

Comment:
Do we really need to listen on ALL events ?
I guess this could costs some performance.